spidermon/contrib/scrapy/monitors.py

```diff
@@ -0,0 +1,145 @@
+# tutorial/monitors.py
+from spidermon import monitors, MonitorSuite, Monitor
+from spidermon.exceptions import NotConfigured
+from ..monitors.mixins.spider import SpiderMonitorMixin
+
+SPIDERMON_MIN_ITEMS = 'SPIDERMON_MIN_ITEMS'
+SPIDERMON_MAX_ERROR = 'SPIDERMON_MAX_ITEMS'
+SPIDERMON_EXPECTED_FINISH_REASONS = 'SPIDERMON_EXPECTED_FINISH_REASONS'
+SPIDERMON_MAX_UNWANTED_HTTP_CODES = 'SPIDERMON_MAX_UNWANTED_HTTP_CODES'
+SPIDERMON_UNWANTED_HTTP_CODES = 'SPIDERMON_UNWANTED_HTTP_CODES'
+
+
+class BaseScrapyMonitor(Monitor, SpiderMonitorMixin):
+    longMessage = False
+
+    def get_settings(self, name, default=None):
+        """ Return a settings prioritized by:
+
+            #. ``{name}`` as a spider attribute if is present
+            #. ``{name}.upper()`` in the ``crawler.settings`` object
+            #. If ``crawler.settings.{name}`` is a dict, try to get the
+               ``{spider.name}`` key of that dictionary, this allows you to
+               have a different configuration per spider in a settings level,
+               like:
+
+               .. code-block:: python
+
+                   SPIDERMON_MAX_ERROR = {
+                        'my_spider': 100,
+                        'other_spider': 10,
+                   }
+
+        """
+        settings_value = self.crawler.settings.get(name.upper())
+        if type(settings_value) == dict:
+            settings_value = settings_value.get(self.spider.name, None)
+        spider_value = getattr(self.spider, name.lower(), None)
```

Comment:
I'm not sure if this a standard practice within scrapy, to default settings to spider attributes.
Mostly because we could achieve the same by just keeping the settings, but using the `custom_settings` _dict_ in the spider.

Reply:
But, if we keep this approach, I think we should document it somewhere.

Reply:
It's documented in the doc string of the function. I can easily change that.
Let's wait for more comments on this one.

Reply:
@raphapassini, I got the docstring, but I was talking about the spidermon docs.
Because this is a feature the API user would need to know without going to the source code.

Reply:
I'm changing this @ejulio but I'm using autodoc, so the docstring _from_ the function *is* going to the documentation :)